Add documentation for http instrumentation using ScopeAgentClient (#24)
undefinedlabs/scope-docs: docs/ios-http-instrumentation.md

<!--END_DOCUSAURUS_CODE_TABS-->

### Using the thin client library

> This same functionality is provided by `ScopeAgentClient`. With this approach, you link your main application with `ScopeAgentClient` and the tests target

Reviewer: ```suggestion
If linking the full `ScopeAgent` library to your main application is not an option, we provide a thin client package called `ScopeAgentClient` that provides the same API described below and communicates with the full `ScopeAgent` library if it is present. With this approach, you link your main application with `ScopeAgentClient` and the tests target with `ScopeAgent`, and the information will be collected during your tests, with minimal impact to your distributable application.
```

with `ScopeAgent`.

Install this dependency to your test targets by adding:

<!--DOCUSAURUS_CODE_TABS-->
<!--Cocoapods-->

```
target 'MyTests' do
 pod 'ScopeAgentClient'
end
```

<!--Carthage-->

```
binary "https://releases.undefinedlabs.com/scope/agents/ios/ScopeAgentClient.json"
```

<!--END_DOCUSAURUS_CODE_TABS-->

## 

## Usage

> If you use ScopeAgent framework use the `SAURLSessionObserver.adapt(_:)` method to modify your `URLRequest` objects. The interface of `SAURLSessionObserver.adapt(_:)` is as follows:

Reviewer: I think putting the code inside a `<!--DOCUSAURUS_CODE_TABS-->` looks better

Author: Done

   public class func adapt(_: URLRequest?) -> URLRequest
   public class func adaptConfiguration(_ : URLSessionConfiguration) -> URLSessionConfiguration
}
```

> Both libraries work in the same way, in the following lines `SAURLSessionObserver` has been used for the examples but `SAURLSessionObserverClient` can be used in the same exact way.

If you application creates its own URLSession or Alamofire `SessionManager` by using a `URLSessionConfiguration`, you can simply call `SAURLSessionObserver.adaptConfiguration(_:)` with the configuration before construction the `URLSession` with it. It will make that all the requests performedn in the `URLSession` just created to add the necessary instrumentation.

<!--DOCUSAURUS_CODE_TABS-->
<!--Alamofire-->

```swift
let configuration = URLSessionConfiguration.default
SAURLSessionObserver.adaptConfiguration(configuration);
let sessionManager = Alamofire.SessionManager(configuration: configuration)
```

<!--URLSession-->

```swift
let configuration = URLSessionConfiguration.default
SAURLSessionObserver.adaptConfiguration(configuration);
let session = URLSession(configuration: configuration)
```

<!--END_DOCUSAURUS_CODE_TABS-->

> ## 

Author: It wasn't (I am using a visual editor and hide it)

  ...
}
```

<!--END_DOCUSAURUS_CODE_TABS-->

## 

> For Alamofire users you can also use a helper class that implements `RequestAdapter` protocol, and using this class session instead of Alamofire default one.

Reviewer: ```suggestion
If you use Alamofire, you can also use a helper class that implements the `RequestAdapter` protocol, and use this class session instead of Alamofire's default one.
```
